query/search/sort.go

```diff
@@ -0,0 +1,90 @@
+package search
+
+import (
+	"github.com/buger/jsonparser"
+	jsoniter "github.com/json-iterator/go"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+)
+
+// TODO: Update this to 3 once https://github.com/typesense/typesense/issues/690 is resolved
+const maxSortOrders = 2
+
+const (
+	ASC  = "$asc"
+	DESC = "$desc"
+)
+
+type Ordering = []SortField
+
+type SortField struct {
+	// Required; Name of field to enable sorting for
+	Name string
+	// Required; True if ascending order is requested, False for descending
+	Ascending bool
+	// Optional; True if missing/empty/null values to be presented at the top of sort order,
+	// else they are sorted to the end by default
+	MissingValuesFirst bool
```

Comment:
We are not accepting this as user input for now, just pushing all empty/missing/null values down the order.